modules/types/include/hephaestus/types/type_formatting.h

```diff
@@ -10,28 +10,63 @@
 
 #include <fmt/chrono.h>
 #include <fmt/format.h>
+#include <magic_enum.hpp>
 #include <range/v3/view/zip.hpp>
 
+#include "hephaestus/utils/concepts.h"
+
 namespace heph::types {
 
 //=================================================================================================
-// Timepoint
+// Vector
 //=================================================================================================
-// A system clock provides access to time and date.
-template <typename T>
-concept IsSystemClock = std::is_same_v<T, std::chrono::system_clock>;
+template <VectorType T>
+[[nodiscard]] inline auto toString(const T& vec) -> std::string {
+  std::stringstream ss;
+
+  const auto indices = std::views::iota(0);
+  const auto indexed_vec = ranges::views::zip(indices, vec);
 
-template <IsSystemClock T>
+  for (const auto& [index, value] : indexed_vec) {
+    const std::string str = fmt::format("  Index: {}, Value: {}\n", index, value);
+    ss << str;
+  }
```

Comment:
Maybe let's add a line at the beginning, otherwise if for example you print two vector in a row it is not clear where the first end and the second start (other than seen the index restart).
It could be:
```
fmt::format("[Vector] size: {}, type: {}", vec.size(), utils::getTypeName<T::type>());
```

Reply:
`toString` is not to be used in an isolated manner, but simply to parse a generic container. Annotating the result happens in the calling function, see e.g. `DummyType`
```
            << "  dummy_vector=" << toString(dummy_type.dummy_vector) << "\n"
```